crates/ruff_python_formatter/src/statement/stmt_with.rs

```diff
@@ -28,7 +26,39 @@ impl FormatNodeRule<StmtWith> for FormatStmtWith {
             ]
         )?;
 
-        if are_with_items_parenthesized(item, f.context())? {
+        // The `with` statement can have one dangling comment on the open parenthesis, like:
+        // ```python
+        // with (  # comment
+        //     CtxManager() as example
+        // ):
+        //     ...
+        // ```
+        //
+        // Any other dangling comments are trailing comments on the colon, like:
+        // ```python
+        // with CtxManager() as example:  # comment
+        //     ...
+        // ```
+        let comments = f.context().comments().clone();
+        let dangling_comments = comments.dangling_comments(item.as_any_node_ref());
+        let partition_point = dangling_comments.partition_point(|comment| {
+            item.items
+                .first()
+                .is_some_and(|with_item| with_item.start() > comment.start())
+        });
+        let (parenthesized_comments, colon_comments) = dangling_comments.split_at(partition_point);
+
+        if !parenthesized_comments.is_empty() {
+            let joined = format_with(|f: &mut PyFormatter| {
+                f.join_with(format_args![text(","), hard_line_break()])
+                    .entries(item.items.iter().formatted())
+                    .finish()
```

Comment:
Does this account for the trailing `,` or should we use `f.join_comma_separated` here, the dangling comments formatting ensures that it renders the group in expanded mode.

Reply:
Ahh good call, it didn't account for that.